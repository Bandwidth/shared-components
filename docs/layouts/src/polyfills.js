'use strict';

if (typeof Promise === 'undefined') {
  // Rejection tracking prevents a common issue where React gets into an
  // inconsistent state due to an error, but it gets swallowed by a Promise,
  // and the user has no idea what causes React's erratic future behavior.
  require('promise/lib/rejection-tracking').enable();
  window.Promise = require('promise/lib/es6-extensions.js');
}

// Object.assign() is commonly used with React.
// It will use the native implementation if it's present and isn't buggy.
Object.assign = require('object-assign');

require('@babel/polyfill');

// ResizeObserver is a polyfill for a tool that subscribes to resize
// events on an element https://wicg.github.io/ResizeObserver/
require('resize-observer');
