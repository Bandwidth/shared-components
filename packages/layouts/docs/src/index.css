body,
html {
  margin: 0;
  min-height: 100%;
}

* {
  box-sizing: border-box;
}
