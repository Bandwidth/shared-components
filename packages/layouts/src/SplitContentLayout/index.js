import React, { createRef } from 'react';
import { RootLayout } from './components';
import { Provider } from './Context';
import MainContent from './MainContent';
import SecondaryContent from './SecondaryContent';
import ActionBar from './ActionBar';
import Modal from './Modal';
import FixedLayer from './FixedLayer';
import throttle from 'lodash.throttle';

export default class SplitContentLayout extends React.PureComponent {
  render() {
    const { children, mainContentLocation, gutter, ...rest } = this.props;

    return (
      <Provider mainContentLocation={mainContentLocation}>
        {(rootElementRef, layerElementRef) => (
          <RootLayout
            innerRef={rootElementRef}
            className="content-layout split-content-layout"
            gutter={gutter}
            {...rest}
          >
            {children}
            <FixedLayer layerRef={layerElementRef} />
          </RootLayout>
        )}
      </Provider>
    );
  }
}

SplitContentLayout.MainContent = MainContent;
SplitContentLayout.SecondaryContent = SecondaryContent;
SplitContentLayout.ActionBar = ActionBar;
SplitContentLayout.Modal = Modal;
