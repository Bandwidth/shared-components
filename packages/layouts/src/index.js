export { default as SplitContentLayout } from './SplitContentLayout';
export { default as RootLayout } from './RootLayout';
