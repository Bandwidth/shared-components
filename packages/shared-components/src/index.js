export {  default as bootstrap } from './bootstrap';
export * from './animations';
export * from './behaviors';
export * from './components/Accordion';
export * from './components/Alert';
export * from './components/Anchor';
export * from './components/BandwidthProvider';
export * from './components/BandwidthThemeProvider';
export * from './components/Breadcrumb';
export * from './components/BulkSelect';
export * from './components/Button';
export * from './components/Callout';
export * from './components/Cards';
export * from './components/Checkbox';
export * from './components/Code';
export * from './components/DatePicker';
export * from './components/DateTimePicker';
export * from './components/DragBox';
export * from './components/DragContainer';
export * from './components/DragGroup';
export * from './components/DragLayer';
export * from './components/DragList';
export * from './components/FileLoader';
export * from './components/Form';
export * from './components/Gutter';
export * from './components/H';
export * from './components/HeaderBar';
export * from './components/HelpText';
export * from './components/Icon';
export * from './components/Input';
export * from './components/Label';
export * from './components/List';
export * from './components/Loader';
export * from './components/Logo';
export * from './components/MethodTag';
export * from './components/Modal';
export * from './components/Money';
export * from './components/Navigation';
export * from './components/NewBadge';
export * from './components/Note';
export * from './components/P';
export * from './components/Pagination';
export * from './components/Radio';
export * from './components/RadioGroup';
export * from './components/SearchBox';
export * from './components/SectionTitle';
export * from './components/Select';
export * from './components/SidebarList';
export * from './components/Spacing';
export * from './components/Steps'
export * from './components/Table';
export * from './components/Tabs';
export * from './components/TextArea';
export * from './components/TimePicker';

// export {  TimeScrubber } from './components/TimeScrubber';
// export {  TimeScrubberContainer } from './components/TimeScrubber/components/TimeScrubberContainer';
// export {  TimeScrubberNudgeButton } from './components/TimeScrubber/components/TimeScrubberNudgeButton';
// export {  TimeScrubberTimeDisplay } from './components/TimeScrubber/components/TimeScrubberTimeDisplay';
// export {  TimeScrubberTimeline } from './components/TimeScrubber/components/TimeScrubberTimeline';
// export {  TimeScrubberViewport } from './components/TimeScrubber/components/TimeScrubberViewport';
// export {  Toast } from './components/Toast';
// export {  Toggle } from './components/Toggle';
// export {  ToggleContainer } from './components/Toggle/styles/ToggleContainer';
// export {  ToggleInput } from './components/Toggle/styles/ToggleInput';
// export {  ToggleLabel } from './components/Toggle/styles/ToggleLabel';
// export {  ToggleButton } from './components/ToggleButton';
// export {  ColorfulButton } from './components/ToggleButton/styles/ColorfulButton';
// export {  SmallButton } from './components/ToggleButton/styles/SmallButton';
// export {  StyledToggleButton } from './components/ToggleButton/styles/StyledToggleButton';
// export {  childrenWithProps } from './extensions/childrenWithProps';
// export {  formatMoney } from './extensions/formatMoney';
// export {  generateId } from './extensions/generateId';
// export {  parseFlex } from './extensions/parseFlex';
// export {  provinces } from './extensions/provinces';
// export {  selectItemPrimaryValue } from './extensions/selectItemPrimaryValue';
// export {  themeGet } from './extensions/themeGet';
// export {  userSpacing } from './extensions/userSpacing';
// export {  userTextSpacing } from './extensions/userTextSpacing';
// export {  layouts } from './layouts';
// export {  Fields } from './layouts/Fields';
// export {  Field } from './layouts/Fields/Field';
// export {  FieldContent } from './layouts/Fields/styles/FieldContent';
// export {  FieldRow } from './layouts/Fields/styles/FieldRow';
// export {  FieldRowContainer } from './layouts/Fields/styles/FieldRowContainer';
// export {  HelpIcon } from './layouts/Fields/styles/HelpIcon';
// export {  LabelContainer } from './layouts/Fields/styles/LabelContainer';
// export {  Flow } from './layouts/Flow';
// export {  FlowItem } from './layouts/Flow/FlowItem';
// export {  FlowRow } from './layouts/Flow/FlowRow';
// export {  fields } from './layouts/Flow/fields';
// export {  FlowCheckbox } from './layouts/Flow/fields/FlowCheckbox';
// export {  FlowFile } from './layouts/Flow/fields/FlowFile';
// export {  FlowInput } from './layouts/Flow/fields/FlowInput';
// export {  FlowMulti } from './layouts/Flow/fields/FlowMulti';
// export {  FlowRadioGroup } from './layouts/Flow/fields/FlowRadioGroup';
// export {  FlowSelect } from './layouts/Flow/fields/FlowSelect';
// export {  FlowTextArea } from './layouts/Flow/fields/FlowTextArea';
// export {  FlowToggle } from './layouts/Flow/fields/FlowToggle';
// export {  FlowItemContainer } from './layouts/Flow/styles/FlowItemContainer';
// export {  FlowItemContent } from './layouts/Flow/styles/FlowItemContent';
// export {  FlowItemFlexibleContent } from './layouts/Flow/styles/FlowItemFlexibleContent';
// export {  FlowItemMoreContent } from './layouts/Flow/styles/FlowItemMoreContent';
// export {  FlowRowStyles } from './layouts/Flow/styles/FlowRowStyles';
// export {  Grid } from './layouts/Grid';
// export {  Horizontal } from './layouts/Horizontal';
// export {  Page } from './layouts/Page';
// export {  Pane } from './layouts/Pane';
// export {  PaneColumn } from './layouts/Pane/PaneColumn';
// export {  PaneRow } from './layouts/Pane/PaneRow';
// export {  PaneSection } from './layouts/Pane/PaneSection';
// export {  PaneSectionContent } from './layouts/Pane/styles/PaneSectionContent';
// export {  PaneSectionTitle } from './layouts/Pane/styles/PaneSectionTitle';
// export {  PaneSectionWrap } from './layouts/Pane/styles/PaneSectionWrap';
// export {  PaneStyles } from './layouts/Pane/styles/PaneStyles';
// export {  SidebarLayout } from './layouts/SidebarLayout';
// export {  theme } from './theme';
// export {  catapultTheme } from './theme/catapultTheme';
// export {  cssvars } from './theme/cssvars';
// export {  index } from './theme/index';
// export {  irisTheme } from './theme/irisTheme';