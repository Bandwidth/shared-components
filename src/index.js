/**
 * THIS IS A GENERATED FILE
 *
 * Please see /tools/generateIndices.js to regenerate after structural changes or component additions / deletions
 */

export { default as bootstrap } from './/bootstrap';
export { default as animations } from './animations';
export { default as Expand } from './animations/Expand';
export { default as behaviors } from './behaviors';
export { default as ExpandToggle } from './behaviors/ExpandToggle';
export { default as Accordion } from './components/Accordion';
export {
  default as AccordionGroup,
} from './components/Accordion/AccordionGroup';
export {
  default as AccordionArrow,
} from './components/Accordion/styles/AccordionArrow';
export {
  default as AccordionBorder,
} from './components/Accordion/styles/AccordionBorder';
export {
  default as AccordionContent,
} from './components/Accordion/styles/AccordionContent';
export {
  default as AccordionGroupContainer,
} from './components/Accordion/styles/AccordionGroupContainer';
export {
  default as AccordionLabel,
} from './components/Accordion/styles/AccordionLabel';
export {
  default as AccordionLabelText,
} from './components/Accordion/styles/AccordionLabelText';
export { default as Alert } from './components/Alert';
export { default as AlertBorder } from './components/Alert/styles/AlertBorder';
export { default as AlertIcon } from './components/Alert/styles/AlertIcon';
export { default as AlertText } from './components/Alert/styles/AlertText';
export { default as Anchor } from './components/Anchor';
export {
  default as ExternalContentAnchor,
} from './components/Anchor/styles/ExternalContentAnchor';
export {
  default as ExternalIconAnchor,
} from './components/Anchor/styles/ExternalIconAnchor';
export {
  default as ExternalTextAnchor,
} from './components/Anchor/styles/ExternalTextAnchor';
export {
  default as InternalContentAnchor,
} from './components/Anchor/styles/InternalContentAnchor';
export {
  default as InternalIconAnchor,
} from './components/Anchor/styles/InternalIconAnchor';
export {
  default as InternalTextAnchor,
} from './components/Anchor/styles/InternalTextAnchor';
export {
  default as BandwidthThemeProvider,
} from './components/BandwidthThemeProvider';
export { default as Breadcrumb } from './components/Breadcrumb';
export { default as Breadcrumbs } from './components/Breadcrumb/Breadcrumbs';
export { default as BulkSelect } from './components/BulkSelect';
export {
  default as BulkSelectBorder,
} from './components/BulkSelect/styles/BulkSelectBorder';
export {
  default as BulkSelectDivider,
} from './components/BulkSelect/styles/BulkSelectDivider';
export {
  default as BulkSelectItem,
} from './components/BulkSelect/styles/BulkSelectItem';
export {
  default as BulkSelectItemContainer,
} from './components/BulkSelect/styles/BulkSelectItemContainer';
export { default as Button } from './components/Button';
export { default as SubmitButton } from './components/Button/SubmitButton';
export { default as Callout } from './components/Callout';
export { default as CalloutTag } from './components/Callout/styles/CalloutTag';
export { default as Cards } from './components/Cards';
export { default as Card } from './components/Cards/Card';
export { default as CardHeader } from './components/Cards/CardHeader';
export { default as CardSection } from './components/Cards/CardSection';
export { default as Checkbox } from './components/Checkbox';
export {
  default as CheckboxContainer,
} from './components/Checkbox/styles/CheckboxContainer';
export {
  default as CheckboxInput,
} from './components/Checkbox/styles/CheckboxInput';
export {
  default as CheckboxLabel,
} from './components/Checkbox/styles/CheckboxLabel';
export { default as Code } from './components/Code';
export { default as CodeBlock } from './components/Code/CodeBlock';
export { default as DatePicker } from './components/DatePicker';
export {
  default as DateRangePicker,
} from './components/DatePicker/DateRangePicker';
export {
  default as DatePickerWrapper,
} from './components/DatePicker/styles/DatePickerWrapper';
export {
  default as DateRangePickerLineSeparator,
} from './components/DatePicker/styles/DateRangePickerLineSeparator';
export {
  default as DateRangePickerWrapper,
} from './components/DatePicker/styles/DateRangePickerWrapper';
export { default as FileLoader } from './components/FileLoader';
export {
  default as FileLoaderDropArea,
} from './components/FileLoader/styles/FileLoaderDropArea';
export {
  default as FileLoaderPreview,
} from './components/FileLoader/styles/FileLoaderPreview';
export { default as Form } from './components/Form';
export { default as FormBox } from './components/Form/FormBox';
export { default as FormColumn } from './components/Form/FormColumn';
export { default as Gutter } from './components/Gutter';
export { default as H } from './components/H';
export { default as H1 } from './components/H/H1';
export { default as H2 } from './components/H/H2';
export { default as H3 } from './components/H/H3';
export { default as H4 } from './components/H/H4';
export { default as H5 } from './components/H/H5';
export { default as HelpText } from './components/HelpText';
export { default as Icon } from './components/Icon';
export { default as icons } from './components/Icon/icons';
export { default as Input } from './components/Input';
export {
  default as InputRevealPasswordWrapper,
} from './components/Input/styles/InputRevealPasswordWrapper';
export { default as InputStyles } from './components/Input/styles/InputStyles';
export { default as Label } from './components/Label';
export { default as List } from './components/List';
export { default as ListItem } from './components/List/ListItem';
export { default as OrderedList } from './components/List/OrderedList';
export { default as UnorderedList } from './components/List/UnorderedList';
export { default as Loader } from './components/Loader';
export {
  default as LoaderContainer,
} from './components/Loader/styles/LoaderContainer';
export { default as LoaderRing } from './components/Loader/styles/LoaderRing';
export { default as Logo } from './components/Logo';
export { default as LogoImage } from './components/Logo/styles/LogoImage';
export { default as MethodTag } from './components/MethodTag';
export { default as Modal } from './components/Modal';
export {
  default as ModalActionContent,
} from './components/Modal/styles/ModalActionContent';
export {
  default as ModalBlocker,
} from './components/Modal/styles/ModalBlocker';
export {
  default as ModalContent,
} from './components/Modal/styles/ModalContent';
export { default as ModalTitle } from './components/Modal/styles/ModalTitle';
export { default as ModalWindow } from './components/Modal/styles/ModalWindow';
export { default as Money } from './components/Money';
export { default as MoneyStyles } from './components/Money/styles/MoneyStyles';
export { default as Navigation } from './components/Navigation';
export {
  default as NavigationTitle,
} from './components/Navigation/NavigationTitle';
export {
  default as NavigationBar,
} from './components/Navigation/styles/NavigationBar';
export {
  default as NavigationHeading,
} from './components/Navigation/styles/NavigationHeading';
export {
  default as NavigationItem,
} from './components/Navigation/styles/NavigationItem';
export {
  default as NavigationItemList,
} from './components/Navigation/styles/NavigationItemList';
export {
  default as NavigationItemListStack,
} from './components/Navigation/styles/NavigationItemListStack';
export {
  default as NavigationLogoPairWrapper,
} from './components/Navigation/styles/NavigationLogoPairWrapper';
export { default as NewBadge } from './components/NewBadge';
export { default as Note } from './components/Note';
export {
  default as NoteContainer,
} from './components/Note/styles/NoteContainer';
export { default as NoteCorner } from './components/Note/styles/NoteCorner';
export { default as Pagination } from './components/Pagination';
export {
  default as PaginationContainer,
} from './components/Pagination/styles/PaginationContainer';
export {
  default as PaginationItem,
} from './components/Pagination/styles/PaginationItem';
export {
  default as PaginationItemPlaceholder,
} from './components/Pagination/styles/PaginationItemPlaceholder';
export { default as Radio } from './components/Radio';
export {
  default as RadioContainer,
} from './components/Radio/styles/RadioContainer';
export { default as RadioInput } from './components/Radio/styles/RadioInput';
export { default as RadioLabel } from './components/Radio/styles/RadioLabel';
export { default as RadioGroup } from './components/RadioGroup';
export { default as RadioButton } from './components/RadioGroup/RadioButton';
export {
  default as RadioGroupButtonContainer,
} from './components/RadioGroup/styles/RadioGroupButtonContainer';
export {
  default as RadioGroupButtonContent,
} from './components/RadioGroup/styles/RadioGroupButtonContent';
export {
  default as RadioGroupButtonInput,
} from './components/RadioGroup/styles/RadioGroupButtonInput';
export {
  default as RadioGroupButtonLabel,
} from './components/RadioGroup/styles/RadioGroupButtonLabel';
export {
  default as RadioGroupButtonLabelText,
} from './components/RadioGroup/styles/RadioGroupButtonLabelText';
export {
  default as RadioGroupContainer,
} from './components/RadioGroup/styles/RadioGroupContainer';
export { default as SearchBox } from './components/SearchBox';
export { default as Suggestion } from './components/SearchBox/Suggestion';
export {
  default as SearchBoxSearchButton,
} from './components/SearchBox/styles/SearchBoxSearchButton';
export {
  default as SearchBoxSuggestionContainer,
} from './components/SearchBox/styles/SearchBoxSuggestionContainer';
export {
  default as SearchBoxSuggestionHighlightText,
} from './components/SearchBox/styles/SearchBoxSuggestionHighlightText';
export {
  default as SearchBoxSuggestionsList,
} from './components/SearchBox/styles/SearchBoxSuggestionsList';
export {
  default as SearchBoxSuggestionsSectionTitle,
} from './components/SearchBox/styles/SearchBoxSuggestionsSectionTitle';
export {
  default as SearchBoxWrapper,
} from './components/SearchBox/styles/SearchBoxWrapper';
export { default as SectionTitle } from './components/SectionTitle';
export { default as Select } from './components/Select';
export {
  default as SelectWrapper,
} from './components/Select/styles/SelectWrapper';
export { default as SidebarList } from './components/SidebarList';
export {
  default as SidebarListItem,
} from './components/SidebarList/SidebarListItem';
export {
  default as SidebarListContainer,
} from './components/SidebarList/styles/SidebarListContainer';
export {
  default as SidebarListItemContainer,
} from './components/SidebarList/styles/SidebarListItemContainer';
export {
  default as SidebarListItemDetails,
} from './components/SidebarList/styles/SidebarListItemDetails';
export {
  default as SidebarListItemLabel,
} from './components/SidebarList/styles/SidebarListItemLabel';
export {
  default as SidebarListShowMore,
} from './components/SidebarList/styles/SidebarListShowMore';
export { default as Spacing } from './components/Spacing';
export { default as Steps } from './components/Steps';
export { default as Step } from './components/Steps/Step';
export {
  default as StepContainer,
} from './components/Steps/styles/StepContainer';
export { default as StepContent } from './components/Steps/styles/StepContent';
export { default as StepList } from './components/Steps/styles/StepList';
export { default as StepNumber } from './components/Steps/styles/StepNumber';
export { default as Table } from './components/Table';
export { default as SimpleTable } from './components/Table/SimpleTable';
export { default as TableControls } from './components/Table/TableControls';
export { default as TableHeader } from './components/Table/TableHeader';
export { default as TableRowDetails } from './components/Table/TableRowDetails';
export { default as TableWrap } from './components/Table/TableWrap';
export { default as TableBody } from './components/Table/styles/TableBody';
export { default as TableCell } from './components/Table/styles/TableCell';
export {
  default as TableControlsContainer,
} from './components/Table/styles/TableControlsContainer';
export {
  default as TableControlsRow,
} from './components/Table/styles/TableControlsRow';
export {
  default as TableHeaderColumnName,
} from './components/Table/styles/TableHeaderColumnName';
export {
  default as TableHeaderSortArrowIcon,
} from './components/Table/styles/TableHeaderSortArrowIcon';
export {
  default as TableHeaderSortArrows,
} from './components/Table/styles/TableHeaderSortArrows';
export {
  default as TableHeaderStyles,
} from './components/Table/styles/TableHeaderStyles';
export {
  default as TableOverlay,
} from './components/Table/styles/TableOverlay';
export { default as TableRow } from './components/Table/styles/TableRow';
export {
  default as TableRowDetailsRow,
} from './components/Table/styles/TableRowDetailsRow';
export {
  default as TableRowDetailsStyles,
} from './components/Table/styles/TableRowDetailsStyles';
export { default as TableStyles } from './components/Table/styles/TableStyles';
export {
  default as TableWrapShadow,
} from './components/Table/styles/TableWrapShadow';
export { default as Tabs } from './components/Tabs';
export { default as Tab } from './components/Tabs/styles/Tab';
export { default as TabContainer } from './components/Tabs/styles/TabContainer';
export { default as TabContent } from './components/Tabs/styles/TabContent';
export { default as TabList } from './components/Tabs/styles/TabList';
export { default as TextArea } from './components/TextArea';
export { default as Toast } from './components/Toast';
export { default as Toggle } from './components/Toggle';
export {
  default as ToggleContainer,
} from './components/Toggle/styles/ToggleContainer';
export { default as ToggleInput } from './components/Toggle/styles/ToggleInput';
export { default as ToggleLabel } from './components/Toggle/styles/ToggleLabel';
export { default as childrenWithProps } from './extensions/childrenWithProps';
export { default as formatMoney } from './extensions/formatMoney';
export { default as generateId } from './extensions/generateId';
export { default as parseFlex } from './extensions/parseFlex';
export { default as provinces } from './extensions/provinces';
export {
  default as selectItemPrimaryValue,
} from './extensions/selectItemPrimaryValue';
export { default as themeGet } from './extensions/themeGet';
export { default as layouts } from './layouts';
export { default as Flow } from './layouts/Flow';
export { default as FlowItem } from './layouts/Flow/FlowItem';
export { default as FlowRow } from './layouts/Flow/FlowRow';
export { default as fields } from './layouts/Flow/fields';
export { default as FlowCheckbox } from './layouts/Flow/fields/FlowCheckbox';
export { default as FlowFile } from './layouts/Flow/fields/FlowFile';
export { default as FlowInput } from './layouts/Flow/fields/FlowInput';
export { default as FlowMulti } from './layouts/Flow/fields/FlowMulti';
export {
  default as FlowRadioGroup,
} from './layouts/Flow/fields/FlowRadioGroup';
export { default as FlowSelect } from './layouts/Flow/fields/FlowSelect';
export { default as FlowTextArea } from './layouts/Flow/fields/FlowTextArea';
export { default as FlowToggle } from './layouts/Flow/fields/FlowToggle';
export {
  default as FlowItemContainer,
} from './layouts/Flow/styles/FlowItemContainer';
export {
  default as FlowItemContent,
} from './layouts/Flow/styles/FlowItemContent';
export {
  default as FlowItemFlexibleContent,
} from './layouts/Flow/styles/FlowItemFlexibleContent';
export {
  default as FlowItemMoreContent,
} from './layouts/Flow/styles/FlowItemMoreContent';
export { default as FlowRowStyles } from './layouts/Flow/styles/FlowRowStyles';
export { default as Page } from './layouts/Page';
export { default as Pane } from './layouts/Pane';
export { default as PaneColumn } from './layouts/Pane/PaneColumn';
export { default as PaneRow } from './layouts/Pane/PaneRow';
export { default as PaneSection } from './layouts/Pane/PaneSection';
export {
  default as PaneSectionContent,
} from './layouts/Pane/styles/PaneSectionContent';
export {
  default as PaneSectionTitle,
} from './layouts/Pane/styles/PaneSectionTitle';
export {
  default as PaneSectionWrap,
} from './layouts/Pane/styles/PaneSectionWrap';
export { default as PaneStyles } from './layouts/Pane/styles/PaneStyles';
export { default as SidebarLayout } from './layouts/SidebarLayout';
export { default as theme } from './theme';
export { default as catapultTheme } from './theme/catapultTheme';
export { default as index } from './theme/index';
export { default as irisTheme } from './theme/irisTheme';
