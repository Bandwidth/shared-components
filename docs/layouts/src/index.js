import React from 'react';
import ReactDom from 'react-dom';

import './index.css';

import { BandwidthProvider } from '@bandwidth/shared-components';
import { HashRouter, Switch, Route } from 'react-router-dom';

import Home from './Home';
import Navigation from './Navigation';
import * as demos from './demos';

import { RootLayout } from '../../src';

ReactDom.render(
  <BandwidthProvider>
    <HashRouter>
      <RootLayout>
        <Route path="/" component={Navigation} />
        <Switch>
          <Route path="/" exact render={Home} />
          <Route path="/root" render={demos.RootLayout} />
          <Route path="/splitContent" component={demos.SplitContent} />
        </Switch>
      </RootLayout>
    </HashRouter>
  </BandwidthProvider>,
  document.getElementById('main'),
);
