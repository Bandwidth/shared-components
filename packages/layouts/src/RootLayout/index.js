import React from 'react';
import { Layout } from './components';
import { ScrollShadow } from '@bandwidth/shared-components';

export default ({ children, scrollShadowProps, ...rest }) => (
  <ScrollShadow global {...scrollShadowProps} {...rest}>
    <Layout>{children}</Layout>
  </ScrollShadow>
);
