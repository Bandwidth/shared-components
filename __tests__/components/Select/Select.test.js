import React from 'react';
import toJSON from 'enzyme-to-json';
import { Select } from '../../../src';

describe('the Select component', () => {
  describe('with string options', () => {
    let wrapper;
    const options = ['foo', 'bar', 'baz'];
    const onChange = jest.fn();

    beforeEach(() => {
      wrapper = shallowWithTheme(
        <Select
          options={options}
          allowNone={false}
          onChange={onChange}
          value="foo"
        />,
      );
    });

    test('renders options correctly', () => {
      expect(
        wrapper.containsAllMatchingElements(
          options.map(option => (
            <option key={option} value={option}>
              {option}
            </option>
          )),
        ),
      ).toBe(true);
    });

    test('selects an option value', () => {
      wrapper
        .find(Select.Container)
        .simulate('change', { target: { value: 'baz' } });
      expect(onChange).toBeCalledWith('baz');
    });
  });

  describe('with complex options and a custom selector', () => {
    let wrapper;
    const options = [
      { custom: 'foo', name: 'bar' },
      { custom: 'bin', name: 'baz' },
      { custom: 'bop', name: 'cor' },
    ];
    const onChange = jest.fn();

    beforeEach(() => {
      wrapper = shallowWithTheme(
        <Select
          options={options}
          allowNone={false}
          onChange={onChange}
          value="foo"
          selectOptionValue={option => option.custom}
          renderOption={option => option.name}
        />,
      );
    });

    test('renders options correctly', () => {
      expect(
        wrapper.containsAllMatchingElements(
          options.map(option => (
            <option key={option.id} value={option.id}>
              {option.name}
            </option>
          )),
        ),
      ).toBe(true);
    });

    test('selects an option value', () => {
      wrapper
        .find(Select.Container)
        .simulate('change', { target: { value: 'bin' } });
      expect(onChange).toBeCalledWith('bin');
    });
  });

  describe('with complex options', () => {
    let wrapper;
    const options = [
      { id: 'foo', name: 'bar' },
      { id: 'bin', name: 'baz' },
      { id: 'bop', name: 'cor' },
    ];
    const onChange = jest.fn();

    beforeEach(() => {
      wrapper = shallowWithTheme(
        <Select
          options={options}
          allowNone={false}
          onChange={onChange}
          value="foo"
        />,
      );
    });

    test('renders options correctly', () => {
      expect(
        wrapper.containsAllMatchingElements(
          options.map(option => (
            <option key={option.id} value={option.id}>
              {option.id}
            </option>
          )),
        ),
      ).toBe(true);
    });

    test('selects an option value', () => {
      wrapper
        .find(Select.Container)
        .simulate('change', { target: { value: 'bin' } });
      expect(onChange).toBeCalledWith('bin');
    });
  });

  describe('with map options', () => {
    let wrapper;
    const options = [
      { get: () => 'foo', name: 'bar' },
      { get: () => 'bin', name: 'baz' },
      { get: () => 'bop', name: 'cor' },
    ];
    const onChange = jest.fn();

    beforeEach(() => {
      wrapper = shallowWithTheme(
        <Select
          options={options}
          allowNone={false}
          onChange={onChange}
          value="foo"
          renderOption={option => option.name}
        />,
      );
    });

    test('renders options correctly', () => {
      expect(
        wrapper.containsAllMatchingElements(
          options.map(option => (
            <option key={option.get()} value={option.get()}>
              {option.name}
            </option>
          )),
        ),
      ).toBe(true);
    });

    test('selects an option value', () => {
      wrapper
        .find(Select.Container)
        .simulate('change', { target: { value: 'bin' } });
      expect(onChange).toBeCalledWith('bin');
    });
  });

  describe('with nonstandard options', () => {
    let wrapper;
    const options = [
      { custom: 'foo', name: 'bar' },
      { custom: 'bin', name: 'baz' },
      { custom: 'bop', name: 'cor' },
    ];
    const onChange = jest.fn();

    beforeEach(() => {
      wrapper = shallowWithTheme(
        <Select
          options={options}
          allowNone={false}
          onChange={onChange}
          value="foo"
        />,
      );
    });

    test('renders options correctly', () => {
      expect(
        wrapper.containsAllMatchingElements(
          options.map(option => (
            <option key={JSON.stringify(option)} value={JSON.stringify(option)}>
              {JSON.stringify(option)}
            </option>
          )),
        ),
      ).toBe(true);
    });

    test('selects an option value', () => {
      wrapper
        .find(Select.Container)
        .simulate('change', { target: { value: 'bin' } });
      expect(onChange).toBeCalledWith('bin');
    });
  });

  describe('with none option', () => {
    let wrapper;
    const options = ['foo', 'bar', 'baz'];
    const noneText = 'Custom';
    const onChange = jest.fn();

    beforeEach(() => {
      wrapper = shallowWithTheme(
        <Select
          options={options}
          onChange={onChange}
          value="foo"
          noneText={noneText}
        />,
      );
    });

    test('renders options correctly', () => {
      const expected = [
        <option key="BW_SHARED_SELECT_NONE" value="BW_SHARED_SELECT_NONE">
          {noneText}
        </option>,
        ...options.map(option => (
          <option key={option} value={option}>
            {option}
          </option>
        )),
      ];
      expect(wrapper.containsAllMatchingElements(expected)).toBe(true);
    });

    test('selects an option value', () => {
      wrapper
        .find(Select.Container)
        .simulate('change', { target: { value: 'BW_SHARED_SELECT_NONE' } });
      expect(onChange).toBeCalledWith(null);
    });
  });
});

describe('the Select component styling', () => {
  test('with string options', () => {
    const wrapper = mountWithTheme(
      <Select options={['foo', 'bar', 'baz']} value="bar" />,
    );
    expect(wrapper).toMatchStyledComponentsSnapshot();
  });
});
