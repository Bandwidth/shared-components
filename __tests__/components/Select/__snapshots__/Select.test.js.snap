// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Select component styling with string options 1`] = `
.cVhWj {
  padding: 15px;
  border: 1px solid #e1e1e1;
  border-radius: 0;
  width: 100%;
  color: #666;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  font-size: 14px;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  line-height: 1.5;
  cursor: pointer;
  box-shadow: none;
  background-color: #fff;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  background-image: url(arrow.png);
  background-repeat: no-repeat;
  background-position: right;
  background-size: 35px;
  -webkit-transition: 0.2s ease all;
  transition: 0.2s ease all;
}

.cVhWj:focus {
  box-shadow: inset 0 -5px 0 #d9f5fc;
  border-color: 1px solid #c2c2c2;
  outline: none;
}

.cVhWj:disabled {
  background: #e1e1e1;
  color: #666;
}

<Select
  __styled-components__={[Function]}
  allowNone={true}
  className={null}
  disabled={false}
  id={null}
  noneText="None"
  options={
    Array [
      "foo",
      "bar",
      "baz",
    ]
  }
  renderOption={[Function]}
  required={false}
  selectOptionValue={[Function]}
  value="bar"
>
  <Select__Styles
    className={null}
    disabled={false}
    id={null}
    onChange={[Function]}
    required={false}
    value="bar"
  >
    <select
      className="Select__Styles-hCutyD cVhWj"
      disabled={false}
      id={null}
      onChange={[Function]}
      required={false}
      value="bar"
    >
      <option
        value="BW_SHARED_SELECT_NONE"
      >
        None
      </option>
      <option
        value="foo"
      >
        foo
      </option>
      <option
        value="bar"
      >
        bar
      </option>
      <option
        value="baz"
      >
        baz
      </option>
    </select>
  </Select__Styles>
</Select>
`;
